src/fdb5/database/Key.cc

```diff
@@ -362,59 +293,22 @@ std::string Key::valuesToString() const {
 }
 
 
-const eckit::StringList& Key::names() const {
+const eckit::StringList& BaseKey::names() const {
     return names_;
 }
 
-std::string Key::value(const std::string& key) const {
+std::string BaseKey::value(const std::string& key) const {
 
     eckit::StringDict::const_iterator it = keys_.find(key);
     ASSERT(it != keys_.end());
     return it->second;
 }
 
-void Key::validateKeysOf(const Key& other, bool checkAlsoValues) const
-{
-    eckit::StringSet missing;
-    eckit::StringSet mismatch;
-
-    const TypesRegistry* registry = canonical_ ? nullptr : &this->registry();
-
-    for (Key::const_iterator j = begin(); j != end(); ++j) {
-        const std::string& keyword = (*j).first;
-        Key::const_iterator k = other.find(keyword);
-        if (k == other.end()) {
-            missing.insert(keyword);
-        }
-        else {
-            if(checkAlsoValues && !canonical_ && !registry->lookupType(keyword).match(keyword, k->second, j->second) ) {
-                mismatch.insert((*j).first + "=" + j->second + " and " + k->second);
-            }
-        }
-    }
-
-    if (missing.size() || mismatch.size()) {
-        std::ostringstream oss;
-
-        if(missing.size()) {
-            oss << "Keywords not used: " << missing << " ";
-        }
-
-        if(mismatch.size()) {
-            oss << "Values mismatch: " << mismatch << " ";
-        }
-
-        oss << "for key " << *this << " validating " << other;
-
-        throw eckit::SeriousBug(oss.str());
-    }
-}
-
-const eckit::StringDict &Key::keyDict() const {
+const eckit::StringDict &BaseKey::keyDict() const {
     return keys_;
 }
 
-metkit::mars::MarsRequest Key::request(std::string verb) const {
+metkit::mars::MarsRequest BaseKey::request(std::string verb) const {
     metkit::mars::MarsRequest req(verb);
```

Comment:
unnecessary copy; `std::string verb` -> `const std::string& verb`
temporary const ref is valid as default (see header)